src/NuGet.Clients/NuGet.PackageManagement.UI/Models/NuGetProjectUpgradeWindowModel.cs

```diff
@@ -48,7 +46,11 @@ private string ProjectName
             {
                 if (string.IsNullOrEmpty(_projectName))
                 {
-                    _projectName = NuGetProject.GetUniqueNameOrName(Project);
+                    _projectName = NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+                    {
+                        return await Project.GetUniqueNameOrNameAsync(CancellationToken.None);
+                    });
```

Comment:
Quoting Andrew here:
"You must not use JTF.Run around access of brokered services. These UI delays can be extreme, and you can deadlock when/if the server can't satisfy the request without a call back to the client when you block the UI thread.
Instead, please refactor to display the UI asynchronously."